CONTRIBUTING.md

```diff
@@ -55,6 +54,48 @@ bridge, you don't need to run these benchmarks.
 If you need any help, the Zinc team hangs out in [sbt/zinc-contrib][].
 Feel free to ask any question.
 
+### Getting familiar with the build
+
+#### Project structure
+
+As of now, the current project structure is not as simple as it can be and we
+believe that it can be simpler in the future. However, as of now, no work is
+happening in this area because it's deemed to have low impact on the overall
+quality of the project.
+
+How is the Zinc build structured? Let's see it.
+
+|Project name| Project description|
```

Comment:
This part should be carefully reviewed. I did my best to explain why these projects are used for, but this structure pre-dates me, so someone has to confirm. Related to https://github.com/sbt/zinc/issues/286.